packages/collector/test/tracing/database/prisma/test.js

```diff
@@ -0,0 +1,212 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+
+'use strict';
+
+const expect = require('chai').expect;
+const fs = require('fs').promises;
+const path = require('path');
+const recursiveCopy = require('recursive-copy');
+const rimraf = require('util').promisify(require('rimraf'));
+
+const supportedVersion = require('@instana/core').tracing.supportedVersion;
+const config = require('../../../../../core/test/config');
+const { retry, verifyHttpRootEntry, verifyExitSpan } = require('../../../../../core/test/test_util');
+const ProcessControls = require('../../../test_util/ProcessControls');
+const globalAgent = require('../../../globalAgent');
+const { executeAsync } = require('../../../test_util/executeCommand');
+
+const providers = ['sqlite', 'postgresql'];
+
+const appDir = __dirname;
+const schemaTargetFile = path.join(appDir, 'prisma', 'schema.prisma');
+const migrationsTargetDir = path.join(appDir, 'prisma', 'migrations');
+
+const mochaSuiteFn = supportedVersion(process.versions.node) ? describe : describe.skip;
+
+mochaSuiteFn('tracing/prisma', function () {
+  this.timeout(Math.max(config.getTestTimeout() * 3, 20000));
+
+  // We need to run npm install for the prisma application (which has its own package.json file) because prisma
+  // downloads a native database driver binary the current platform (Darwin, Linux, ...) as an npm postinstall hook.
+  before(async () => {
+    await executeAsync('npm install', appDir);
```

Comment:
_Question_: Why do we not put the dependency `@prisma/client'` into our root package.json? We already have many native dependencies in their.

Reply:
There is a comment in line 31 that explains it. In addition to that particular reason, we also need to run `npm run generate` (that is, `prisma generate`) for each provider and Prisma is quite finicky with paths, the generate command seems to basically expect `prisma/schema.prisma` in the projects root folder.

I'll add some more comments to explain this.

Reply:
On second thought, the comment is indeed misleading. It is less about the native drivers (as you correctly pointed out), but only about `prisma generate` not working if the dependency is not installed locally here for this test app.

Reply:
I've expanded the comment for this.